Singularity can't run Clear Linux

## Version of Singularity:

3.0.3

### Expected behavior

The ability to run and shell into the container.

This is attempting to create Clear Linux containers using the Singularity runtime. So far I've attempted two methods, building directly from Docker and manually adding in missing files through a definition.

### Actual behavior

Singularity errors out.

Straight from Docker:
```text
$ singularity build --sandbox intel docker://clearlinux:latest
WARNING: passwd file doesn't exist in container, not updating
WARNING: group file doesn't exist in container, not updating
```

Definition File:
```text
=====
bootstrap: docker
from: clearlinux:latest

%post
    touch /etc/passwd
    touch /etc/group
=====

$ singularity shell intel
panic: runtime error: index out of range

goroutine 8 [running, locked to thread]:
github.com/sylabs/singularity/internal/pkg/util/fs/files.Passwd(0xc000366780, 0x2e, 0xc00034e240, 0xc, 0x3e8, 0xc0002db400, 0x0, 0x200, 0x0, 0x0)
    /home/mroche/rpmbuild/BUILD/singularity-3.0.3/gopath/src/github.com/sylabs/singularity/internal/pkg/util/fs/files/passwd.go:49 +0x860
github.com/sylabs/singularity/internal/pkg/runtime/engines/singularity.(*container).addIdentityMount(0xc0000b4240, 0xc0003347a0, 0x0, 0x0)
    /home/mroche/rpmbuild/BUILD/singularity-3.0.3/gopath/src/github.com/sylabs/singularity/internal/pkg/runtime/engines/singularity/container.go:1592 +0x60d
github.com/sylabs/singularity/internal/pkg/runtime/engines/singularity.(*container).addIdentityMount-fm(0xc0003347a0, 0x0, 0x0)
    /home/mroche/rpmbuild/BUILD/singularity-3.0.3/gopath/src/github.com/sylabs/singularity/internal/pkg/runtime/engines/singularity/container.go:103 +0x34
github.com/sylabs/singularity/internal/pkg/util/fs/mount.(*System).MountAll(0xc0003347a0, 0x9, 0x0)
    /home/mroche/rpmbuild/BUILD/singularity-3.0.3/gopath/src/github.com/sylabs/singularity/internal/pkg/util/fs/mount/system.go:78 +0x341
github.com/sylabs/singularity/internal/pkg/runtime/engines/singularity.create(0xc0002d16f0, 0xc000334760, 0xf98, 0x7f7045c0c0b8, 0xc00000e9c8)
    /home/mroche/rpmbuild/BUILD/singularity-3.0.3/gopath/src/github.com/sylabs/singularity/internal/pkg/runtime/engines/singularity/container.go:151 +0x74d
github.com/sylabs/singularity/internal/pkg/runtime/engines/singularity.(*EngineOperations).CreateContainer(0xc0002d16f0, 0xf98, 0xa77cc0, 0xc00000e9c8, 0x0, 0x0)
    /home/mroche/rpmbuild/BUILD/singularity-3.0.3/gopath/src/github.com/sylabs/singularity/internal/pkg/runtime/engines/singularity/create.go:86 +0x3f3
github.com/sylabs/singularity/internal/app/starter.Master.func1(0x6, 0xc00020afc0, 0xc000201d80, 0xf98)
    /home/mroche/rpmbuild/BUILD/singularity-3.0.3/gopath/src/github.com/sylabs/singularity/internal/app/starter/master.go:46 +0x185
created by github.com/sylabs/singularity/internal/app/starter.Master
    /home/mroche/rpmbuild/BUILD/singularity-3.0.3/gopath/src/github.com/sylabs/singularity/internal/app/starter/master.go:36 +0x1a3
```

### Steps to reproduce behavior

Create a Clear Linux container by any means.

--- a/internal/pkg/util/fs/files/files_linux_test.go
+++ b/internal/pkg/util/fs/files/files_linux_test.go
@@ -7,6 +7,7 @@ package files
 
 import (
 	"bytes"
+	"io/ioutil"
 	"os"
 	"testing"
 
@@ -28,6 +29,19 @@ func TestGroup(t *testing.T) {
 	if err != nil {
 		t.Errorf("should have passed with correct group file")
 	}
+	// with an empty file
+	f, err := ioutil.TempFile("", "empty-group-")
+	if err != nil {
+		t.Error(err)
+	}
+	emptyGroup := f.Name()
+	defer os.Remove(emptyGroup)
+	f.Close()
+
+	_, err = Group(emptyGroup, uid, gids)
+	if err != nil {
+		t.Error(err)
+	}
 }
 
 func TestPasswd(t *testing.T) {
@@ -43,6 +57,19 @@ func TestPasswd(t *testing.T) {
 	_, err = Passwd("/etc/passwd", "/home", uid)
 	if err != nil {
 		t.Errorf("should have passed with correct passwd file")
+	}
+	// with an empty file
+	f, err := ioutil.TempFile("", "empty-passwd-")
+	if err != nil {
+		t.Error(err)
+	}
+	emptyPasswd := f.Name()
+	defer os.Remove(emptyPasswd)
+	f.Close()
+
+	_, err = Passwd(emptyPasswd, "/home", uid)
+	if err != nil {
+		t.Error(err)
 	}
 }
 

--- a/internal/pkg/util/fs/files/group.go
+++ b/internal/pkg/util/fs/files/group.go
@@ -65,8 +65,8 @@ func Group(path string, uid int, gids []int) (content []byte, err error) {
 		return content, fmt.Errorf("failed to read group file content in container: %s", err)
 	}
 
-	if content[len(content)-1] != '\n' {
-		content = append(content, byte('\n'))
+	if len(content) > 0 && content[len(content)-1] != '\n' {
+		content = append(content, '\n')
 	}
 
 	for _, gid := range groups {

--- a/internal/pkg/util/fs/files/passwd.go
+++ b/internal/pkg/util/fs/files/passwd.go
@@ -46,8 +46,8 @@ func Passwd(path string, home string, uid int) (content []byte, err error) {
 	}
 	userInfo := fmt.Sprintf("%s:x:%d:%d:%s:%s:%s\n", pwInfo.Name, pwInfo.UID, pwInfo.GID, pwInfo.Gecos, homeDir, pwInfo.Shell)
 
-	if content[len(content)-1] != '\n' {
-		content = append(content, byte('\n'))
+	if len(content) > 0 && content[len(content)-1] != '\n' {
+		content = append(content, '\n')
 	}
 
 	sylog.Verbosef("Creating template passwd file and appending user data: %s\n", path)

